lib/rules/no-extra-parens.js

```diff
@@ -431,6 +435,10 @@ module.exports = {
                     return;
                 }
 
+                if (node.body.type === "ConditionalExpression" && ENFORCE_FOR_ARROW_CONDITIONALS) {
+                    return;
+                }
```

Comment:
No, if `enforceForArrowConditionals` is true, then I would expect the rule to enforce that there are no extra parentheses, so it should report that case. If `enforceForArrowConditionals` is false, then the rule should not report the extra parens.

So the following examples are valid:

```js
/* eslint no-extra-parens: [error, { enforceForArrowConditionals: true }] */

var a = (b) => 1 ? 2 : 3;
```

```js
/* eslint no-extra-parens: [error, { enforceForArrowConditionals: false }] */

var a = (b) => 1 ? 2 : 3;
```

```js
/* eslint no-extra-parens: [error, { enforceForArrowConditionals: false }] */

var a = (b) => (1 ? 2 : 3);
```

The following example is invalid:

```js
/* eslint no-extra-parens: [error, { enforceForArrowConditionals: true }] */

var a = (b) => (1 ? 2 : 3);
```